### PR TITLE
Energy usage

### PR DESCRIPTION
**Pull request by nhassink**
- added measure_power capability
- added meter_power capability
- fixed bug in available / unavailable logic which would lead to
permanently unavailable plugs, used anna logic

### CODE REVIEW

File: drivers/stretch/driver.js [hunk 1]
```diff
@@ -283,6 +283,38 @@ module.exports.capabilities = {
 				callback(true, false);
 			}
 		}
+	},
+	measure_power: {
+		get: function (device_data, callback) {
+			if (!device_data) callback(true, null);
```
Comment (RobinBol) [inline drivers/stretch/driver.js]: Generally, it is best practice to return a callback statement, so `if (!device_data) return callback(true, null);`. In this case it is also really necessary, image the case that device_data is not provided, then the callback will return an error as it should, but the rest of the function (trying to find a device and fetch its value), will also continue, which could result in crash. This is not a big issue, as it should never be the case, but just a tip for next time.

File: drivers/stretch/driver.js [hunk 2]
```diff
@@ -317,6 +349,20 @@ function listenForEvents(device_data) {
 
 			// Emit realtime
 			module.exports.realtime(device_data_obj, "onoff", onoff);
+	
+		}).on("measurepowerchanged", function (measure_power) {
```
Comment (RobinBol) [inline drivers/stretch/driver.js]: Something like `measure_power_changed` would be more readable in my opinion, but now I am nitpicking.

Comment (nhassink) [inline drivers/stretch/driver.js]: I agree

File: drivers/stretch/driver.js (hunk 1 above)
Comment (nhassink) [inline drivers/stretch/driver.js]: I understand, but I copy pasted and modified the one-off capability. this should have the same bug then

**Comment by nhassink** [comment]
Thanks for the quick approval. Will try to find some testers

**Comment by nhassink** [comment]
Found 1 other plugwise user so far who has tested the changes and found no issues.

**Comment by RobinBol** [comment]
App update is released, thanks again:)

**Pull request merged**